### PR TITLE
Runway-work

### PR DESCRIPTION
**Pull request by Tyler-R-Kendrick**
added tests
Addressed pylint feedback
created devcontainer definition
added launch settings

### CODE REVIEW

**Review by petemessina — DISMISSED** [review]
LGTM

File: backend/main.py [hunk 1]
```diff
@@ -1,58 +1,68 @@
...
-load_dotenv(override=True, dotenv_path=f"{os.getcwd()}/.env")
-
 from models.chat_request import Chat
+from models.chat_response import to_response_item
 from models.vector_store_options import VectorStoreOptions
-from models.openai_options import OpenAIOptions
+from models.openai_options import OpenAIOptions, ModelOptions, ApiOptions
 from approaches.multi_index_chat_builder import MultiIndexChatBuilder
-from approaches.chat_conversation import ChatConversation
+from approaches.chat_conversation import ChatConversationOptions, chat
```
Comment (jaspecla) [inline backend/main.py:13]: I'm a little wary of having an object called Chat and a method called chat.

File: backend/approaches/chat_conversation.py [hunk 2]
```diff
@@ -1,61 +1,77 @@
...
+class ChatConversationOptions:
+    """Class used to manage the chat conversation
+        and chain runnables together for the chat conversation"""
+    system_prompt: str
+    default_return_message: str
 
-        return general_chain
+def _route(options: ChatConversationOptions,
+    builder: MultiIndexChatBuilder,
+    info):
```
Comment (jaspecla) [inline backend/approaches/chat_conversation.py]: What is "info"?

**Review by jaspecla — CHANGES_REQUESTED** [review]
A few variable names that I feel are unclear (see comments).

File: backend/main.py (hunk 1 above)
Comment (Tyler-R-Kendrick) [reply backend/main.py:13]: fair.

File: backend/approaches/chat_conversation.py (hunk 2 above)
Comment (Tyler-R-Kendrick) [reply backend/approaches/chat_conversation.py]: not sure myself - point made haha

**Review by jaspecla — APPROVED** [review]
Changes addressed, LGTM

**Review by petemessina — APPROVED** [review]

**Pull request merged**